lldb/source/Commands/CommandOptionsProcessLaunch.cpp

```diff
@@ -38,7 +38,18 @@ Status CommandOptionsProcessLaunch::SetOptionValue(
   case 's': // Stop at program entry point
     launch_info.GetFlags().Set(eLaunchFlagStopAtEntry);
     break;
-
+  case 'm': // Stop at main function
+  {
+    TargetSP target_sp =
+        execution_context ? execution_context->GetTargetSP() : TargetSP();
+    BreakpointSP bp_sp = target_sp->CreateBreakpoint(
+        nullptr, nullptr, "main", eFunctionNameTypeAuto, eLanguageTypeUnknown,
+        0, eLazyBoolNo, false, false);
+    if (!bp_sp)
+      error.SetErrorString("Breakpoint creation failed.\n");
+    launch_info.SetShell(HostInfo::GetDefaultShell());
```

Comment:
remove this line. We shouldn't be modifying the shell in the entry point breakpoint case.